projects/js-packages/charts/src/components/legend/hooks/use-chart-legend-items.ts

```diff
@@ -105,41 +115,48 @@ function processSeriesData(
 /**
  * Processes point data into legend items
  * @param pointData          - The point data to process
- * @param theme              - The chart theme for colors
+ * @param getElementStyles   - Function to get element styles
  * @param showValues         - Whether to show values in legend
  * @param legendValueDisplay - What type of value to display
  * @param withGlyph          - Whether to include glyph rendering
  * @param glyphSize          - Size of the glyph
  * @param renderGlyph        - Component to render the glyph
+ * @param legendShape        - The shape to use for the item
  * @return Array of processed legend items
  */
 function processPointData(
 	pointData: ( DataPointDate | DataPointPercentage )[],
-	theme: ChartTheme,
+	getElementStyles: ( params: GetElementStylesParams ) => ElementStyles,
 	showValues: boolean,
 	legendValueDisplay: LegendValueDisplay,
 	withGlyph: boolean,
 	glyphSize: number,
-	renderGlyph?: < Datum extends object >( props: GlyphProps< Datum > ) => ReactNode
+	renderGlyph?: < Datum extends object >( props: GlyphProps< Datum > ) => ReactNode,
+	legendShape?: LegendShape< SeriesData[], number >
 ): BaseLegendItem[] {
 	const mapper = ( point: DataPointDate | DataPointPercentage, index: number ) => {
-		const baseItem = {
+		const { color, glyph, shapeStyles } = getElementStyles( {
+			data: point as DataPointPercentage,
+			index,
+			legendShape,
+		} );
+
+		const baseItem: BaseLegendItem = {
 			label: point.label,
 			value: formatPointValue( point, showValues, legendValueDisplay ),
-			color: ( point as DataPointPercentage ).color ?? theme.colors[ index % theme.colors.length ],
-			group: ( point as DataPointPercentage ).group,
-			index,
-			overrideColor: ( point as DataPointPercentage ).color,
+			color,
+			shapeStyle: shapeStyles,
 		};
 
-		if ( withGlyph && renderGlyph ) {
-			const itemWithGlyph = {
-				...baseItem,
-				glyphSize,
-				renderGlyph,
-			};
-
-			return itemWithGlyph;
+		if ( withGlyph ) {
+			const glyphToUse = renderGlyph || glyph;
```

Comment:
This support for legend glyphs isn't working yet

Reply:
Fixed in [81a34b8](https://github.com/Automattic/jetpack/pull/45207/commits/81a34b8be0004913f958670592c7675484a6c1fc)